state/upgrades_test.go

```diff
@@ -2926,6 +2931,43 @@ func (s *upgradesSuite) TestUpdateInheritedControllerConfig(c *gc.C) {
 	)
 }
 
+func (s *upgradesSuite) TestEnsureDefaultModificationStatus(c *gc.C) {
+	coll, closer := s.state.db().GetRawCollection(statusesC)
+	defer closer()
+
+	s.makeMachine(c, life.Alive)
+	s.makeMachine(c, life.Dying)
+
+	expected := bsonMById{
+		{
+			"_id":        "m#0#modification",
+			"model-uuid": "",
+			"status":     "idle",
+			"statusinfo": "",
+			"statusdata": bson.M{},
+			"neverset":   false,
+			"updated":    int64(1),
+		},
+	}
+
+	c.Log(pretty.Sprint(expected))
+	s.assertUpgradedDataWithFilter(c, EnsureDefaultModificationStatus,
+		bson.D{{"_id", bson.RegEx{"#modification$", ""}}},
+		expectUpgradedData{coll, expected},
+	)
+}
+
+func (s *upgradesSuite) makeMachine(c *gc.C, life life.Value) {
+	coll, closer := s.state.db().GetRawCollection(machinesC)
+	defer closer()
+
+	err := coll.Insert(bson.M{
+		"machineid": "0",
```

Comment:
This doesn't insert a valid machine, there is no model-uuid specified, and the resulting test above is showing that the "_id" and model-uuid are not set correctly.